scripts/west_commands/boards.py

```diff
@@ -68,6 +71,18 @@ def do_run(self, args, _):
         else:
             name_re = None
 
+        modules_board_roots = []
+
+        for module in zephyr_module.parse_modules(ZEPHYR_BASE):
+            try:
+                board_root = module.meta['build']['settings']['board_root']
+            except KeyError:
+                continue
+
+            modules_board_roots.append(Path(module.project) / board_root)
```

Comment:
Personally I like when code is handling such cases itself and not relying on catching errors.

As example, the following would be an alternative:
```python
            build = module.meta.get('build', {'settings': {'board_root': None}})
            board_root = build.get('settings', {'board_root': None}).get('board_root')

            if board_root is not None:
                modules_board_roots.append(Path(module.project) / board_root)
```

@mbolivar-nordic any thoughts ?

Reply:
Didn't test, but maybe:
```
module.meta.get('build', {}).get('settings', {}).get('board_root')
```
will do the job?

Reply:
> I think this is kind of personal taste here. 

it sure is.

Personally, I prefer the try-catch for instances where thing may happen, but are not expected to happen in a correctly configured system.
And in this part, a missing `build` or `settings` key is perfectly valid, even in correctly configured system.



> module.meta.get('build', {}).get('settings', {}).get('board_root')

Did some brief testing and it does seem to work just the same, and much cleaner.

Reply:
@mniestroj ping

Reply:
Got rid of try-catch in favor of `module.meta.get('build', {}).get('settings', {}).get('board_root')`.